src/main/java/com/sleekbyte/tailor/listeners/TodoCommentListener.java

```diff
@@ -0,0 +1,54 @@
+package com.sleekbyte.tailor.listeners;
+
+import com.sleekbyte.tailor.common.Location;
+import com.sleekbyte.tailor.common.Messages;
+import com.sleekbyte.tailor.common.Rules;
+import com.sleekbyte.tailor.output.Printer;
+import com.sleekbyte.tailor.utils.ListenerUtil;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.Token;
+
+import java.util.List;
+
+/**
+ * Class to analyze TODO_SYNTAX comments.
+ */
+public final class TodoCommentListener extends CommentAnalyzer {
+    /**
+     * Create instance of CommentAnalyzer.
+     *
+     * @param tokenStream Token stream obtained from lexer
+     * @param printer     An instance of Printer
+     */
+    public TodoCommentListener(CommonTokenStream tokenStream, Printer printer) {
+        super(tokenStream, printer);
+    }
+
+    @Override
+    public void analyze() {
+        checkTodoSyntaxInSingleLineComments();
+        checkTodoSyntaxInMultiLineComments();
+    }
+
+    private void checkTodoSyntaxInSingleLineComments() {
+        String todoRegex = "(?s)// TODO: [^\\s].+";
```

Comment:
`// TODO(adithya): something goes here` is pretty common among developers and organizations (both Coursera and Google used this), so I think we should allow for that